backend/infrahub/graphql/auth/query_permission_checker/object_permission_checker.py

```diff
@@ -0,0 +1,66 @@
+from infrahub.auth import AccountSession
+from infrahub.core import registry
+from infrahub.core.account import ObjectPermission
+from infrahub.core.branch import Branch
+from infrahub.core.constants import PermissionDecision
+from infrahub.database import InfrahubDatabase
+from infrahub.exceptions import PermissionDeniedError
+from infrahub.graphql import GraphqlParams
+from infrahub.graphql.analyzer import InfrahubGraphQLQueryAnalyzer
+from infrahub.utils import extract_camelcase_words
+
+from .interface import CheckerResolution, GraphQLQueryPermissionCheckerInterface
+
+
+class ObjectPermissionChecker(GraphQLQueryPermissionCheckerInterface):
+    account_session: AccountSession
+
+    async def supports(self, db: InfrahubDatabase, account_session: AccountSession, branch: Branch) -> bool:
+        self.account_session = account_session
```

Comment:
Ah yes good catch! The real problem here is that these would be global objects and if we reuse a session in this way there's a chance that we'd read this `self._account_session` in within the event loop if another user is also online...

I'll change it so that check also requires the account session parameter. I think that's better than using an account_id as a string as we might have other checks that would care about how the user was authenticated.